File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length<=1){
            return nums.length;
        }
        Set<Integer> set=new HashSet<>();
        for(int i=0;i<nums.length;i++){
            if(!set.contains(nums[i])){
                set.add(nums[i]);
            }
        }
        int count=0;
        int max=1;
        for(Integer num :set){
            if(!set.contains(num-1)){
                while(set.contains(num)){
                    count++;
                    num++;
                }
                max=Math.max(max,count);
            }
            count=0;
        }
        return max;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        myset=set()
        max=0
        count=0
        l=0
        for r in s:
            while r in myset:
                myset.remove(s[l])
                l=l+1
                count=count-1
            myset.add(r)
            count=count+1
            if count > max :
                max=count
        
        return max
            
